### DISCUSSION TITLE
Make const implicitly typed

### DISCUSSION THREAD

**Discussion by hdadr** [dotnet/csharplang#4195] [Language Ideas] [answered]
Just like how [var](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/var) works, i would like to use const similar way.

For example:
const int x = 10;  -> 
const x = 10;

const double pi = 3.14; ->
const pi = 3.14;

Edit: The purpose of this idea is to eleminate the unnecessary type declaration.

**Comment by JanTamis** [comment]
Isn't it a better idea to allow the keyword var in const?

**Reply by hdadr** [reply]
The idea behind making const implicitly typed is to eliminate type declaration which i find unnecessary. 
I feel var also unnecessary here.

**Reply by JanTamis** [reply]
I think it would make the code less readable.

**Reply by hdadr** [reply]
Coming from typescript background, for me using this way eleminates some "noises", so it makes more readable.

**Comment by svick** [comment, marked as answer]
This is a duplicate of the championed proposal https://github.com/dotnet/csharplang/issues/106.
